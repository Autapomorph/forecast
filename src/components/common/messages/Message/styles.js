import styled from 'styled-components';

export const StyledMessageWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 10px;
  border-radius: 5px;
`;

export const StyledMessageContent = styled.span`
  text-align: center;
  font-size: 1.5rem;
`;
