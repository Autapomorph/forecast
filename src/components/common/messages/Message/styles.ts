import styled from 'styled-components/macro';

export const StyledMessageWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 10px;
  border-radius: 5px;
  font-size: 1.3rem;
`;

export const StyledMessageContent = styled.span`
  text-align: center;
`;
