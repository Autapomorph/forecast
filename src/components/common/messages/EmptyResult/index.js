import React from 'react';

import { StyledEmptyResultMessage } from './styles';

const EmptyResult = ({ children, ...props }) => (
  <StyledEmptyResultMessage {...props}>{children}</StyledEmptyResultMessage>
);

export default EmptyResult;
