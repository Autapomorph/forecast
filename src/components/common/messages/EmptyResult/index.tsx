import React from 'react';

import * as S from './styles';

const EmptyResult: React.FC = ({ children, ...props }): React.ReactElement => (
  <S.EmptyResultMessage {...props}>{children}</S.EmptyResultMessage>
);

export default EmptyResult;
