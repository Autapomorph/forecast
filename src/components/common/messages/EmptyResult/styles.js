import styled from 'styled-components';

import Message from '../Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledEmptyListMessage = styled(Message)``;
