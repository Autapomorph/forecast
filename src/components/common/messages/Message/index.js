import React from 'react';

import { StyledMessageWrapper, StyledMessageContent } from './styles';

const Message = ({ children, ...props }) => (
  <StyledMessageWrapper {...props}>
    <StyledMessageContent>{children}</StyledMessageContent>
  </StyledMessageWrapper>
);

export default Message;
