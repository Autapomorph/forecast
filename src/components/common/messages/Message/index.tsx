import React from 'react';

import { StyledMessageWrapper, StyledMessageContent } from './styles';

const Message: React.FC = ({ children, ...props }): React.ReactElement => (
  <StyledMessageWrapper {...props}>
    <StyledMessageContent>{children}</StyledMessageContent>
  </StyledMessageWrapper>
);

export default Message;
