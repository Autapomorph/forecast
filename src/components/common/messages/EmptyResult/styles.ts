import styled from 'styled-components';

import Message from 'components/common/messages/Message';

// eslint-disable-next-line import/prefer-default-export
export const StyledEmptyResultMessage = styled(Message)`
  margin: 20px 0;
`;
