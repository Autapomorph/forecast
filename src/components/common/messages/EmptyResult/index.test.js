import EmptyResult from '.';
import Message from '../Message';

describe('EmptyResult tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<EmptyResult />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render defined text', () => {
    const text = 'text';
    const wrapper = mount(<EmptyResult>{text}</EmptyResult>);

    expect(wrapper.find(Message).text()).toEqual(text);
  });
});
