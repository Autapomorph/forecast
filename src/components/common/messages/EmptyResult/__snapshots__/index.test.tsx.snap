// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyResult tests renders properly 1`] = `ShallowWrapper {}`;
